components/properties/src/props.rs

```diff
@@ -41,6 +49,147 @@ enum EnumeratedProperty {
 }
 
+/// Private marker type for PropertyValueNameToEnumMapper
+/// to work for all properties at once
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub(crate) struct ErasedValueNameMap;
+impl DataMarker for ErasedValueNameMap {
+    type Yokeable = PropertyValueNameToEnumMapV1<'static>;
+}
+
+/// A struct capable of looking up a property value from a string name
+///
+/// The name can be a short name (`Lu`), a long name(`Uppercase_Letter`),
+/// or an alias.
+///
+/// Property names can be looked up using "strict" matching (looking for a name
+/// that matches exactly), or "loose matching", where the name is allowed to deviate
+/// in terms of ASCII casing, whitespace, underscores, and hyphens.
+///
+/// # Example
+///
+/// ```
+/// use icu::properties::GeneralCategory;
+///
+/// let lookup = GeneralCategory::lookup_values(&icu_testdata::unstable())
+///                  .expect("The data should be valid");
+/// // short name for value
+/// assert_eq!(lookup.get_strict("Lu"), Some(GeneralCategory::UppercaseLetter));
+/// assert_eq!(lookup.get_strict("Pd"), Some(GeneralCategory::DashPunctuation));
+/// // long name for value
+/// assert_eq!(lookup.get_strict("Uppercase_Letter"), Some(GeneralCategory::UppercaseLetter));
+/// assert_eq!(lookup.get_strict("Dash_Punctuation"), Some(GeneralCategory::DashPunctuation));
+/// // name has incorrect casing
+/// assert_eq!(lookup.get_strict("dashpunctuation"), None);
+/// // loose matching of name
+/// assert_eq!(lookup.get_loose("dash-punctuation"), Some(GeneralCategory::DashPunctuation));
+/// // fake property
+/// assert_eq!(lookup.get_strict("Animated_Gif"), None);
+/// ```
+pub struct PropertyValueNameToEnumMapper<T> {
+    map: DataPayload<ErasedValueNameMap>,
+    marker: PhantomData<fn() -> T>,
+}
+
+impl<T: TrieValue> PropertyValueNameToEnumMapper<T> {
+    /// Get the property value as a u16, doing a strict search looking for
+    /// names that match exactly
+    #[inline]
+    pub fn get_strict_u16(&self, name: &str) -> Option<u16> {
+        get_strict_u16(&self.map, name)
+    }
+
+    /// Get the property value as a `T`, doing a strict search looking for
+    /// names that match exactly
+    #[inline]
+    pub fn get_strict(&self, name: &str) -> Option<T> {
+        self.get_strict_u16(name)
+            .and_then(|u| T::try_from_u32(u as u32).ok())
+    }
+
+    /// Get the property value as a u16, doing a loose search looking for
+    /// names that match case-insensitively, ignoring ASCII hyphens, underscores, and
+    /// whitespaces.
+    #[inline]
+    pub fn get_loose_u16(&self, name: &str) -> Option<u16> {
+        get_loose_u16(&self.map, name)
+    }
+
+    /// Get the property value as a `T`, doing a loose search looking for
+    /// names that match case-insensitively, ignoring ASCII hyphens, underscores, and
+    /// whitespaces.
+    #[inline]
+    pub fn get_loose(&self, name: &str) -> Option<T> {
```

Comment:
Suggestion: ICU uses the terminology "lenient" and "strict"

Reply:
Yeah but PropertyValueAliases.txt calls it "loose", and that file is more speccy. @markusicu ?